client/src/components/UpdateLink.tsx

```diff
@@ -17,28 +14,32 @@ interface Props {
 }
 
 //TODO: rename this component to LinkUpdateBtn
-export function MoreDetails(props: Props) {
+export function UpdateLink(props: Props) {
   const [token, setToken] = useState('');
 
   const { generateLinkToken, linkTokens } = useLink();
-  // creates new link token for each item in bad state
-  useEffect(() => {
+
+  const initiateLinkUpdate = async () => {
     generateLinkToken(props.userId, props.itemId); // itemId is set because link is in update mode
-  }, [props.userId, props.itemId, generateLinkToken]);
+  };
 
   useEffect(() => {
     setToken(linkTokens.byItem[props.itemId]);
   }, [linkTokens, props.itemId]);
 
   return (
     <div className="more-details">
-      {token != null && token.length > 0 && (
-        <LinkButton userId={props.userId} itemId={props.itemId} token={token}>
+      {
+        <Button small centered inline onClick={initiateLinkUpdate}>
+          {' '}
```

Comment:
is this something we could use CSS to create the spacing for instead?

Reply:
deleted this...this was a prettier thing